File: 1323_maximum_69_number.py
class Solution:
    def maximum69Number (self, num: int | str) -> int:
        num = str(num)
        if num.count('9') == len(num):
            return int(num)
        
        i = 0
        while i < len(num) and num[i] == '9':
            i += 1
        return int(num[:i] + '9' + num[i + 1:])

File: JavaScript/1323_maximum_69_number.js
/**
 * @param {number} num
 * @return {number}
 */
var maximum69Number = function(num) {
    let first_six_index = -1;
    let digits = String(num).split('');
    for (const [i, digit] of digits.entries()) {
        if (digit === '6') {
            first_six_index = i;
            break;
        }
    }

    if (first_six_index === -1) {
        return num;
    } else {
        digits[first_six_index] = '9';
        return parseInt(digits.join(""));
    }
};

